test/routs/testSoldierRouts.js

```diff
@@ -71,12 +79,194 @@ describe("Soldiers tests", function () {
         "stam": "lol",
         "limitations": ["yeshiva"]
       }));
-      Http.onreadystatechange = (e) => {
-        if (Http.readyState == 4 && Http.status == 200) {
-          expect(Http.responseText).to.eql("One or more fields is invalid");
-          done();
+      Http.onreadystatechange = (connetErr) => {
+        if (connetErr) {
+          done(connetErr);
+        } else {
+          if (Http.readyState == 4 && Http.status == 200) {
+            expect(Http.responseText).to.eql("One or more fields is invalid");
+            done();
+          }
+        }
+      }
+    })
+  });
+
+  describe("Soldiers get test", function () {
+
+    it("Should be able to return correct respone when the soldier path doesn't contain id", function (done) {
```

Comment:
There is no way this test should be this long, I should not need to read 26 lines of code for a single POST request to the db.